apps/engine.c

```diff
@@ -48,28 +48,38 @@ const OPTIONS engine_options[] = {
 
 static int append_buf(char **buf, int *size, const char *s)
 {
-    if (*buf == NULL) {
-        *size = 256;
-        *buf = app_malloc(*size, "engine buffer");
-        **buf = '\0';
-    }
+    const int expand = 256;
+    unsigned int len = strlen(s) + 1;
+    char *p = *buf;
+
+    if (p == NULL) {
+        *size = (len + expand - 1) / expand * expand;
+        p = *buf = app_malloc(*size, "engine buffer");
+    } else {
+        const unsigned int blen = strlen(p);
+
+        if (blen > 0)
+            len += 2 + blen;
+
+        if (len > (unsigned int)*size) {
```

Comment:
len is unsigned which means a warning if not.  Is there a recommended way to deal with this?